test/easyconfigs/easyconfigs.py

```diff
@@ -1422,7 +1422,11 @@ def template_easyconfig_test(self, spec):
                 # 4th value is toolchain spec
                 if len(dumped_dep) >= 4:
                     if len(orig_dep) >= 4:
-                        self.assertEqual(dumped_dep[3], orig_dep[3])
+                        # A SYSTEM toolchain is dumped as True
+                        if orig_dep[3] == {'version': 'system', 'name': 'system'}:
+                            self.assertEqual(dumped_dep[3], True)
+                        else:
+                            self.assertEqual(dumped_dep[3], orig_dep[3])
```

Comment:
The change implemented in https://github.com/easybuilders/easybuild-framework/pull/4069 will make this no longer necessary (but the reverse is needed to avoid that the easyconfig test suite will start failing for all easyconfigs which are still using `True` to indicate that a dependency should use the system toolchain, cfr. https://github.com/easybuilders/easybuild-easyconfigs/pull/16126)